src/chapters/shell-intro/walkthrough_bash-basics.rst

```diff
@@ -0,0 +1,1043 @@
+
+.. sourcecode:: bash
+   :caption: Linux/BASH
+
+   $ <command> | grep '<search term>' <output from command>
+
+We can *pipe* the output of ``ls`` (directory contents as a string) as the string input used by ``grep`` to filter just the results we need. Our pipeline would look like this:
+
+.. sourcecode:: bash
+
+   $ ls --> dir contents string | --> grep 'search term' <dir contents string> --> search results string
+
+What if we wanted to check for details about the ``dotnet`` program by using the long form ``ls`` output:
+
+.. sourcecode:: bash
+   :caption: Linux/BASH
+
+   $ ls -l /usr/bin | grep 'dotnet'
+   lrwxrwxrwx 1 root   root           22 May 20 15:37 dotnet -> ../share/dotnet/dotnet
+
+You can pipe to and from many CLI programs thanks to the standard use of strings as outputs and inputs. As a final example let's search through the help output of ``dotnet``. If you were to view the help output directly you would end up scrolling through many lines.
+
+What if you just want to know how to publish a project (something we will soon cover)? We can use piping to automate the process of searching through the lines manually:
+
+.. sourcecode:: bash
+   :caption: Linux/BASH
+
+   $ dotnet --help | grep 'publish'
+   publish           Publish a .NET project for deployment.
+
+Scripting
+=========
+
+Shell scripting is the process of automating a series of commands. The key to automation is to understand the logical steps needed to perform a task manually. In this course we will use scripting to automate operational tasks like provisioning and managing cloud resources on Azure. 
+
+Early in the course we will provide you with scripts that you will be encouraged to read but not expected to write. After getting comfortable with the manual steps you will learn how to write and use your own scripts. 
+
+Essentials
+----------
+
+Commands
+^^^^^^^^
+
+You can use any command in a script that you are able to issue manually in the Shell REPL. Unlike the REPL which will prompt you for the next command, scripts are written in a file with each independent command or statement occupying a single line.
+
+A statement, like other languages, is an independent instruction like defining a variable or constructing a loop. We distinguish these from commands which refer to actual CLI programs like those you would call from the Shell REPL.
+
+.. admonition:: note
+
+   When providing code samples for scripts we will remove the ``$`` character used in REPL examples. Each separate line is its own command or statement.
+
+Script File Extensions
+^^^^^^^^^^^^^^^^^^^^^^
+
+Because file extensions are arbitrary in Linux, a script file can have any extension (or none at all). However, it is customary to use the ``.sh`` extension as a note to signify that the script should be interpreted as BASH commands.
+
+Comments
```

Comment:
I would move this waaay up in the article since you use comments regularly in the shell sections. It's true they are seen more often in scripts, they are comments in the shell REPL as well.